### DISCUSSION TITLE
Setting x and y variables when using a pandas dataframe

### DISCUSSION THREAD

**Discussion by pranavlal** [winedarksea/AutoTS#176] [Q&A]
Hi all,

I discovered auto-ts yesterday. I have checked the documentation and am confused. 

I have a pandas dataframe that has several features and also contains the variable to be predicted. The features are a series of transformations to stock prices as well as the outputs of running technical indicators on said prices. The dataframe is indexed by  date.

How do I specify the various features and the variable to be predicted?

Pranav

**Comment by winedarksea** [comment]
This is 'auto' forecasting so it will create various features automatically. I just won the M6 competition stock picking decisions category with AutoTS using simply the automated features. 
It sounds like your data is pretty much ready to go. You can just plug in the entire data frame and the multivariate models will automatically use the other series. You can see the examples in the readme.md and production_example, replacing the load_daily(long=False) sample with your own dataframe. It might be worth passing `weights` to tell if you don't care so much about forecasting the indicators, and the prices are the main focus. Also if you happen to know the future values of the technical indicators, then you should use a second dataframe to `future_regressor` both fit and predict which contains your indicators. Feel free to ask more specific questions here if you run into confusion.

**Comment by pranavlal** [comment]
Hi,
Many thanks for your answer. 
My initial thought was to implement a model such that I would feed in hourly data, some features and use the returns  as what I was predicting. I have historical data of returns. So, my X value would be a time indexed series with values like rsi, differences in prices etc. 
The returns are just the percentage change  in prices. If the returns were greater than 0, then I would generate 1 as the signal. 
When I would deploy the system, I would grab live data every hour and feed that to the model. If I would get a signal of 1, I would then take a long trade. 
I have other sets of conditions to close the position. 

If I understand your answers correctly, you have done a lot of this automatically. Could you perhaps explain how a live trading system would work? When I feed in live data, would the model give me the price in the next hour such that I could make a decision whether to buy or not?

**Reply by winedarksea** [reply]
I should say that even having put lots of time into forecasting and having won a competition for stock picking with it, I _still don't use it to manage my own money_. It's just very risky and uncertain.

Daily pricing is going to be a bit easier than hourly. The compute time and data size is the issue with hourly data. You can look at the [`production_example.py`](https://github.com/winedarksea/AutoTS/blob/master/production_example.py) in this repo for an example of a live forecasting setup, including stock prices there, but no decision making (set drop_most_recent=0 and make sure the data is align to the most recent day manually). You should really think about how you will use the probabilistic bounds (upper and lower forecasts) along with the median point forecast in make your decision, if you want to minimize the risk.
